src/main/scala/cognite/spark/v1/DataModelInstancesRelation.scala

```diff
@@ -291,98 +287,169 @@ object DataModelInstanceRelation {
       case a => throw new CdfSparkException(unknownPropertyTypeMessage(a))
     }
 
-  private def jsonFromDouble(num: Double): Json =
-    Json.fromDouble(num).getOrElse(throw new CdfSparkException(s"Numeric value $num"))
-  private def tryGetValue(propertyType: String, nullable: Boolean): PartialFunction[Any, Json] = // scalastyle:off
+  private def toDateProperty: Any => DateProperty = {
+    case x: LocalDate => DateProperty(x)
+    case x: java.sql.Date => DateProperty(x.toLocalDate)
+    case x: LocalDateTime => DateProperty(x.toLocalDate)
+    case a => throw new CdfSparkException(notValidPropertyTypeMessage(a, "date"))
+  }
+
+  private def toTimestampProperty: Any => TimeStampProperty = {
+    case x: Instant =>
+      TimeStampProperty(OffsetDateTime.ofInstant(x, ZoneId.systemDefault()).toZonedDateTime)
+    case x: java.sql.Timestamp =>
+      // Using ZonedDateTime directly without OffSetDateTime, adds ZoneId to the end of encoded string
+      // 2019-10-02T07:00+09:00[Asia/Seoul] instead of 2019-10-02T07:00+09:00, API does not like it.
+      TimeStampProperty(OffsetDateTime.ofInstant(x.toInstant, ZoneId.systemDefault()).toZonedDateTime)
+    case a => throw new CdfSparkException(notValidPropertyTypeMessage(a, "timestamp"))
```

Comment:
It is nice, but we shouldn't use the system default `ZoneId`, rather convert it to UTC explicitly.
Maybe `LocalDateTime.from(ts.toLocalDateTime.atZone(ZoneOffset.UTC))` or similar. Tests would be needed to make sure we don't get it wrong for different time zones.

Reply:
`TimeStampProperty` wants `ZonedDateTime` as value type. I'll see if it can be adapted to that.

Reply:
Well, this PR is quite big already, so it might be good to get this merged first even with a simple-but-not-always-correct solution for timestamps first, and then add tests and fix edge cases in another (shorter) PR.

Reply:
Sure I think this version can be merged 👍 I can create another PR to test type conversions and filters on timestamp etc.